pilot/cmd/pilot-agent/options/agent.go

```diff
@@ -66,6 +66,7 @@ func NewAgentOptions(proxy *model.Proxy, cfg *meshconfig.ProxyConfig) *istioagen
 		ProxyDomain:                 proxy.DNSDomain,
 		IstiodSAN:                   istiodSAN.Get(),
 		DualStack:                   features.EnableDualStack,
+		ExitIfSDSSocketNotFound:     exitIfSDSSocketNotFoundEnv,
```

Comment:
Based on context in https://github.com/istio/istio/issues/45534#issuecomment-1631201311 - can I propose we just change this flag to `UseExternalWorkloadSDS`?

If it's set, we check for the alt socket and error if it's not there. If it isn't we don't.